src/renderer/api/kube-api.ts

```diff
@@ -357,8 +366,96 @@ export class KubeApi<T extends KubeObject = any> {
     });
   }
 
-  watch(): () => void {
-    return kubeWatchApi.subscribeApi(this);
+  watch(opts: KubeApiWatchOptions = { namespace: "" }): () => void {
+    if (!opts.abortController) {
+      opts.abortController = new AbortController();
+    }
+    const { abortController, namespace, callback } = opts;
+
+    // FIXME
+    const watchUrl = `http://${this.kind}-watch.${window.location.host}${apiKubePrefix}${this.getWatchUrl(namespace)}`;
```

Comment:
I'm out of ideas how to customise request host without passing full url. There must be a better way.

Reply:
This is needed as a workaround for browser connection limits. It seems that chromium allows only 6 or so connections to a host. If we generate different subdomain for each watched kind we can overcome this limitation.